### PR TITLE
WIP: Issue 749 improve admin interface: tag area

### PR DESCRIPTION
**Pull request by netagonen**
This relates to https://github.com/rubymonsters/speakerinnen_liste/issues/789. 
Please review :). 
Frontend styling is still needed.
One problem I couldn't figure out yet:
Merging a tag does NOT delete the merged tag, BUT the test for that passes: spec/controllers/admin/tag_controller_spec.rb:213
Same behavior is also on `master` (there at spec/controllers/admin/tag_controller_spec.rb:176)
help?

### CODE REVIEW

**Comment by tyranja** [comment]
Thank you @netagonen! When I find some time I will look at it tomorrow.

**Comment by tyranja** [comment]
@netagonen I tried reproduce the behaviour of merging of tags. 
I used the code in the app for merging of the tag and in my case it worked:

```
rails c
existing_tag=ActsAsTaggableOn::Tag.where(name: "familie").first
@tag=ActsAsTaggableOn::Tag.where(name: "familiengründung auf freundschaftlicher basis").first
existing_tag.merge(@tag)
```

when I look at the old tag ( `@tag` ) I get an empty relation:
```
ActsAsTaggableOn::Tag.where(name: "familiengründung auf freundschaftlicher basis")
=> #<ActiveRecord::Relation []>
```
and the new tag has one `taggings_count` more:

```
ActsAsTaggableOn::Tag.where(name: "familie")
=> #<ActiveRecord::Relation [#<ActsAsTaggableOn::Tag id: 524, name: "familie", taggings_count: 11>]>
```

**Comment by tyranja** [comment]
@netagonen look really good :+1: I tested it and added some css for the styling. 

I like it that you can see the profiles that are tagged.

Maybe you can add also some sorting order. Like in profile so we can see the tag that is the newest.
sort

What do you think?

**Comment by tyranja** [comment]
I have deployed the branch to staging: https://staging-speakerinnen-liste.herokuapp.com/

Do you have a admin account there for testing?

**Comment by netagonen** [comment]
I will happily add a sorting to the tags overview. 
I don't have credentials for the staging env I think.

**Comment by tyranja** [comment]
@netagonen Just sign in on staging; https://staging-speakerinnen-liste.herokuapp.com/ and I will set your account as an admin and that's it. :)

**Comment by netagonen** [comment]
Hi @tyranja ,
1. I created an account on staging with my email: netagonen@gmail.com
2. About the sorting: the tags overview table has at the moment the following columns- name, categories, languages and profiles. I will add a "created at" column. I think it makes sense to have the sorting on the name and date columns only, because the other columns could have multiple values. What do you think?

**Comment by tyranja** [comment]
@netagonen 

1) Ok, you are an admin now on staging. 

2). sounds good! thank you

Once the tests are green, we could merge the branch and push it to production. What do you think?

**Comment by netagonen** [comment]
Sure! Still have more work on the tags area but I will make a new branch for it

**Comment by tyranja** [comment]
Hello @netagonen I changed one wording. and I wonder if we should validate that tags for the presence of profiles. You know, we had this isse 2 weeks ago and didn't know why. And in the end the tags belonged to no profile and that was the error.

So maybe we should ad something like:

`  validates :profiles, :presence => true` to the actsastaggable initializer?

You could ad that in the next branch, if you think that makes sense. I would deploy this branch later today, when you have no objection :)

**Comment by netagonen** [comment]
Sure, I think the validation makes sense, but it might be more complicated than that - the profile comes from the tagging, not the tag. So I think it could be done in the next branch :)
Thanks

**Comment by tyranja** [comment]
@netagonen if we take the creation date from the **first** tagging from the tag then we should have the creating date of the tag.

File: app/views/admin/tags/index.html.erb [hunk 1]
```diff
@@ -15,48 +15,55 @@
...
-              category_id: category.id, page: params[:page], q: params[:q],
-              uncategorized: params[:uncategorized]), method: :post %>
+            <% tag.taggings.each do |tagging| %>
+              <%= link_to tagging.taggable.name_or_email, admin_profile_path(tagging.taggable),
+                  html_options = { title: t(:show, scope: 'admin.dashboard.profiles.edit') } %><br/>
+            <% end %>
+          </td>
+          <td>
+            <% if  !tag.taggings.empty? %>
+              <%= tag.taggings.sort_by{|t| t.created_at }.last.created_at.strftime('%d.%m.%Y') %>
```
Comment (tyranja) [inline app/views/admin/tags/index.html.erb]: `ActsAsTaggableOn::Tag.first.taggings.first.created_at` I would change the last tagging to first tagging to get the creation date of the tag.

Comment (netagonen) [reply app/views/admin/tags/index.html.erb]: Changed tagging to be the first one. I think for this pull request everything is ready

**Comment by netagonen** [comment]
BTW, twice after pushing to origin I had this failure, but one push after all tests were green. 
screenshot from 2018-12-04 14-06-25

File: app/views/admin/tags/index.html.erb (hunk 1 above)
Comment (tyranja) [reply app/views/admin/tags/index.html.erb]: Thank you!!!

**Comment by tyranja** [comment]
Thanks a lot! That was great work!

**Comment by tyranja** [comment]
That failure happens sometimes :woman_shrugging:

**Pull request merged**